src/app/frontend/deploy/deploy_stateconfig.js

```diff
@@ -44,3 +46,23 @@ function resolveNamespaces($resource) {
 
   return resource.get().$promise;
 }
+
+/**
+ * @param {!angular.$resource} $resource
+ * @return {!angular.Resource<!backendApi.ReplicaSetSpec>}
+ * @ngInject
+ */
+function getProtocolsResource($resource) {
+  return $resource('/api/appdeployments/protocols');
+}
+
+/**
+ * @param {!angular.Resource<!backendApi.Protocols>} protocolsResource
+ * @returns {!angular.$q.Promise}
+ * @ngInject
+ */
+function getDefaultProtocols(protocolsResource) {
+  /** @type {!backendApi.ProtocolsSpec} */
+  let protocolsSpec = {isExternal: false};
```

Comment:
Should default isExternal value be injected to the controller? I say yes. It should be easy to add. At least, create a constant for it.